chemprop/train/make_predictions.py

```diff
@@ -293,11 +300,19 @@ def predict_and_save(
                         datapoint.row[pred_name + f"_model_{idx}"] = pred
 
         # Save
-        with open(args.preds_path, "w") as f:
-            writer = csv.DictWriter(f, fieldnames=full_data[0].row.keys())
-            writer.writeheader()
-            for datapoint in full_data:
-                writer.writerow(datapoint.row)
+        if train_args.is_atom_bond_targets:
+            data = {}
+            for key in full_data[0].row.keys():
+                data[key] = [datapoint.row[key] for datapoint in full_data]
+            preds_dataframe = pd.DataFrame.from_dict(data)
+            preds_dataframe.to_pickle(args.preds_path)
```

Comment:
Could we support saving the results both as .pkl or as csv? And maybe detect whether the extension provided is csv or pickle to choose between them?